app/controllers/UpscanController.scala

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.ApplicationConfig
+import connectors.UpscanInitiateConnector
+import models.{File, UploadResponse}
+import play.api.Logging
+import play.api.mvc._
+import services.{FilesSessionService, SessionCacheService}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.auth.DefaultAuthConnector
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class UpscanController @Inject()(upscanInitiateConnector: UpscanInitiateConnector,
+                                     val authConnector: DefaultAuthConnector,
+                                     val filesSessionService: FilesSessionService,
+                                     val sessionService: SessionCacheService,
+                                     val mcc: MessagesControllerComponents,
+                                     implicit val appConfig: ApplicationConfig,
+                                     fileUploadView: views.html.file_upload,
+                                     fileUploadSuccessView: views.html.file_upload_successful,
+                                     cannotUploadAnotherView: views.html.cannot_upload_another_file
+																		) extends FrontendController(mcc) with RasController with PageFlowController with Logging {
+
+	implicit val ec: ExecutionContext = mcc.executionContext
+
+  def get: Action[AnyContent] = Action.async {
```

Comment:
this is a very large method, honestly reading it I got quite confused. I think we should pull this method apart a bit and make some smaller methods just to make it more readable. I had a quick go let me know what you think...

Reply:
See latest changes